src/QsCompiler/CompilationManager/AssemblyLoader.cs

```diff
@@ -98,13 +106,18 @@ public static bool LoadReferencedAssembly(Uri asm, out References.Headers header
         public static bool LoadSyntaxTree(
             Stream stream,
             [NotNullWhen(true)] out QsCompilation? compilation,
+            SyntaxTreeSerializationProtocol serializationProtocol = SyntaxTreeSerializationProtocol.NewtonsoftBinary,
```

Comment:
There shouldn't be a default value here, since the right protocol depends on the stream, so whoever gets the stream needs to know the protocol.

Reply:
But looking more closely at the code, I don't thing the two ways of loading should be unified in the same method. I think LoadSyntaxTree should be the loading via Bond (i.e. it should always be the loading of what is currently generated by the compiler). If we for now want to support processing old dlls then I would add a separate method for loading the old version. It may make sense to consider if this method should be taking something else than a stream, in which case both methods could have the same name but a different first argument.

Reply:
I went with you recommendation and created another `LoadSyntaxTree` method that receives a byte array as first argument. Should we mark the one that receives a stream as deprecated?

Reply:
Yes, I think that makes sense.

Reply:
Deprecated the API with the appropriate message.